analysis/quantifier-comparisons/12-spatial-transcriptomics-storage.Rmd

```diff
@@ -0,0 +1,206 @@
+---
+title: "Storing Spatial Transcriptomics Results"
+author: "Ally Hawkins for CCDL"
+output: 
+  html_notebook:
+    toc: true
+    toc_float: true
+---
+
+In this notebook we are doing a quick exploration of the use of storing spatial data in a [`SpatialExperiment` object](https://bioconductor.org/packages/release/bioc/vignettes/SpatialExperiment/inst/doc/SpatialExperiment.html#3_Common_operations) or in a [`Seurat` object](https://satijalab.org/seurat/articles/spatial_vignette.html). 
+We will take a look at one library, SCPCR000372, that has been quantified with Spaceranger and compare the use of `SpatialExperiment` to `Seurat`.  
+
+## Set Up
+
+```{r}
+library(magrittr)
+library(ggplot2)
+library(SingleCellExperiment)
+library(SpatialExperiment)
+```
+
+```{r}
+# load in benchmarking functions that will be used for copying data and generating sample tables
+function_path <- file.path(".." ,"benchmarking-functions", "R")
+file.path(function_path, list.files(function_path, pattern = "*.R$")) %>%
+  purrr::walk(source)
+```
+
+```{r}
+# set up file paths 
+base_dir <- here::here()
+
+# output folder to store alevin-fry and spaceranger quants from S3
+data_dir <- file.path(base_dir, "data", "spatial") 
+s3_out <- file.path(data_dir, "data", "quants")
+
+# results directory 
+results_dir <- file.path(data_dir, "results")
+
+# create directories
+if(!dir.exists(data_dir)){
+  dir.create(data_dir, recursive = TRUE)
+}
+
+if(!dir.exists(results_dir)){
+  dir.create(results_dir, recursive = TRUE)
+}
+```
+
+
+```{r}
+mito_file <- file.path(base_dir, "sample-info", "Homo_sapiens.GRCh38.103.mitogenes.txt")
+  
+# read in mito genes 
+mito_genes <- readr::read_tsv(mito_file, col_names = "gene_id")
+mito_genes <- mito_genes %>%
+  dplyr::pull(gene_id) %>%
+  unique()
+```
+
+## Copy data from S3
+
+```{r}
+# grab alevin fry and cellranger output
+sample_id <- c("SCPCR000372")
+
+aws_copy_samples(local_dir = s3_out,
+                 s3_dir = "s3://nextflow-ccdl-results/scpca",
+                 samples = sample_id,
+                 tools = c("alevin-fry-knee", "cellranger"))
+```
```

Comment:
One thing that I noticed earlier: it might be nice to split these into separate chunks, with the download only happening if run manually, for easier rerunning of the notebook as a whole:
~~~suggestion
```{r}
# set samples to process
sample_id <- c("SCPCR000372")
```

```{r eval=FALSE}
# download alevin fry and cellranger output
aws_copy_samples(local_dir = s3_out,
                 s3_dir = "s3://nextflow-ccdl-results/scpca",
                 samples = sample_id,
                 tools = c("alevin-fry-knee", "cellranger"))
```
~~~